lib/devise/passkeys/controllers/concerns/reauthentication_challenge.rb

```diff
@@ -31,6 +31,14 @@ def passkey_reauthentication_challenge_session_key
             "#{resource_name}_current_reauthentication_challenge"
           end
 
+          # This method is responsible for generating the the object class for Devise resource name
+          # that will be used to find the authentication_keys configured in Devise
+          #
+          # @return [Object] The Devise resource name, as an object
+          def resource_class
+            Object.const_get(resource_name.to_s.camelize)
```

Comment:
@tcannonfodder : I needed to have `User` as a class to pull the Devise configuration for this resource, but I am not sure if using Object.const_get is a good idea here or not.

I know this info must come from the resource class because you can bypass the Devise config at the resource class level (using `:authentication_keys => [:email]`).

Reply:
Also, I've picked the location for this method based on the changes that I needed to do.
But maybe there is a better location for it.

Reply:
Upon reflection, I'm not sure we actually need to add this method. Devise provides a `resource_class` method out of the box in `DeviseController`, see: https://github.com/heartcombo/devise/blob/9f80dc2562524f744e8633b8562f2a0114efb32b/app/controllers/devise_controller.rb#L50

In the template repo, we have `Users::ReauthenticationController` as a subclass of `DeviseController`, which is what we want in this case. Since it's related to authentication, it should 100% be a subclass of the core `DeviseController`.

Reply:
Indeed! Thanks for pointing it out.

Removing this method all together runs the test suite just fine!